### PR TITLE
Add MySQL8-based tests (with percona-8.0) & Upgrade GitHub Actions to Latest Versions (in Preparation for Node 16 EOL)

### LINKED ISSUE #46: `docker-compose.yaml`에 서비스 추가 (MySQL version 8 테스트 추가)
[`pymysqlreplication.tests.test_data_type.test_partition_id`](https://github.com/julien-duponchelle/python-mysql-replication/blob/45f823fa0ea6cd19de84b39aa78f5f181fafe18b/pymysqlreplication/tests/test_data_type.py#L694-L708) 테스트가 의도대로 작동하기 위해서는 MySQL 8.0.16 버전 이상의 컨테이너가 필요합니다.

해당 테스트에 대한 상세한 내용은 [PR](https://github.com/julien-duponchelle/python-mysql-replication/pull/355)을 참고해주세요

### LINKED ISSUE #89: Upgrade GitHub Actions to Latest Versions (in Preparation for Node 16 EOL)
#46 

## Overview
This PR updates GitHub Actions to use the latest versions of actions/checkout and actions/setup-python in preparation for the [end-of-life of Node 16 in September 2023.](https://nodejs.org/en/blog/announcements/nodejs16-eol)

## Description
The default runtime of Node 16 will reach its end-of-life on 11 Sep 2023. To adapt to this change and benefit from the features and optimizations in the newer GitHub Actions versions

## Change
Updates GitHub Actions Versions:
From actions/checkout@v2 to actions/checkout@v4
From actions/setup-python@v2 to actions/setup-python@v4

## Benefits
- Keeps the Actions Up-to-date: Utilizes performance improvements and new features from the latest versions.
- Future-proofs the Repository: Prepares the codebase for the end of support for Node 16.

## References
- [actions/checkout Release Notes](https://github.com/actions/checkout/releases)
- [actions/setup-python Release Notes](https://github.com/actions/setup-python/releases)

### PR DESCRIPTION
**Pull request by heehehe**
Resolve #46 Resolve #89

#### Add MySQL-8 based test (with percona-8.0)
Since below two features need to test based on MySQL version 8,
this PR added percona-8.0 docker and test based on MySQL version 8.
- Optional Metadata (https://github.com/julien-duponchelle/python-mysql-replication/pull/446)
- partition info (https://github.com/julien-duponchelle/python-mysql-replication/pull/355)

#### Update github actions version
This PR also updates GitHub Actions to use the latest versions of actions/checkout and actions/setup-python
in preparation for the [end-of-life of Node 16 in September 2023.](https://nodejs.org/en/blog/announcements/nodejs16-eol)
The default runtime of Node 16 will reach its end-of-life on 11 Sep 2023.
To adapt to this change and benefit from the features and optimizations in the newer GitHub Actions versions.

#### Add MySQL-8 based test (with percona-8.0)
- Add percona-8.0 docker image with port 3309
- Add `PyMySQLReplicationVersion8TestCase` class in `pymysqlreplication/tests/base.py`
- Change TestOptionalMetaData's inheritance 
from `base.PyMySQLReplicationTestCase` to `base.PyMySQLReplicationVersion8TestCase`
- Add `TestDataTypeVersion8` class in `pymysqlreplication/tests/test_data_type.py`

#### Update github actions version
- From actions/checkout@v2 to actions/checkout@v4
- From actions/setup-python@v2 to actions/setup-python@v4

#### Add MySQL-8 based test (with percona-8.0)
- Could test which only works on Mysql verison 8.

#### Update github actions version
- Keeps the Actions Up-to-date: Utilizes performance improvements and new features from the latest versions.
- Future-proofs the Repository: Prepares the codebase for the end of support for Node 16.

#### Update github actions version
- [actions/checkout Release Notes](https://github.com/actions/checkout/releases)
- [actions/setup-python Release Notes](https://github.com/actions/setup-python/releases)

### CODE REVIEW

File: docker-compose-test.yml [hunk 1]
```diff
@@ -37,6 +37,15 @@ services:
...
 
+  percona-8.0:
+    <<: *mysql
+    image: percona:8.0
+    platform: linux/amd64
+    ports:
+      - "3309:3306"
+    networks:
+      - default
+
```
Comment (sean-k1) [inline docker-compose-test.yml:48]: environment:
      MYSQL_ALLOW_EMPTY_PASSWORD: "yes" 
      환경에 이거 안넣어줘도 잘돌아가는게 신기하네요..

**Review by starcat37 — COMMENTED** [review]
버전 관련은 잘 모르기도 해서 팔로업을 못했네요😭 고생하셨어요!🥹🥹

**Review by dongwook-chan — APPROVED** [review]
역시 3조답게 깔끔하게 수정해주셨네요 
제 테스트 살려주신 것도 감사하네요 
생명의 은인이십니다 🙏🏻🙏🏻🙏🏻🙏🏻

File: docker-compose-test.yml (hunk 1 above)
Comment (mjs1995) [reply docker-compose-test.yml:48]: [앵커](https://docs.docker.com/compose/compose-file/10-fragments/)를 썼던데 신기하더라고요!

**Review by mikaniz — APPROVED** [review]
수고 많으셨습니다!! 경험이 없는 부분이라 도움을 못 드려서 죄송하네요.. 😢
항상 많이 배우고 있습니다!

**Comment by heehehe** [comment]
https://github.com/julien-duponchelle/python-mysql-replication/pull/484 반영 완료

**Pull request closed without merging**